File: src/zork/Zork.java
package zork;

import zork.engine.ZorkLoop;
import zork.utils.Parser;
import java.util.Scanner;
import zork.generators.MapGenerator;
import zork.generators.PlayerGenerator;

/**
 * Main class which instantiates the core Objects of the Program like the parser,
 * the generators, the game state etc.
 *
 * @author d.peters
 * @version 1.0
 */
public class Zork {
  /**
   * Scanner to scan user input from console.
   */
  private final Scanner scanner;

  /**
   * The parser object to parse input commands.
   */
  private final Parser parser;

  /**
   * Generator for player objects.
   */
  private final PlayerGenerator pGen;

  /**
   * Generator for maps (currently all the none player objects are also
   * created here).
   */
  private final MapGenerator mGen;

  /**
   * State of the game (all game objects and their data are stored here).
   */
  private final GameState game;

  /**
   * Game actions.
   */
  private final ZorkActions actions;

  /**
   * This object contains the main program loop.
   */
  private final ZorkLoop loop;

  /**
   * Default constructor. Initializes the scanner object, the loop to start
   * and end the game as well the Parser and the actual game objects
   */
  public Zork() {
    this.scanner = new Scanner(System.in);
    this.parser = new Parser(this.scanner);
    this.pGen = new PlayerGenerator(this.scanner);
    this.mGen = new MapGenerator();
    this.game = new GameState(this.pGen.createPlayer(), this.mGen.createRooms());
    this.game.setCurrentRoom(this.game.getRooms().get(3));
    this.actions = new ZorkActions(this.parser, this.game);
    this.loop = new ZorkLoop(this.actions);
  }

  /**
   * Main method instantiates the game object and starts its loop.
   *
   * @param args the command line arguments
   */
  public static void main(String[] args) {
    Zork game = new Zork();
    game.loop.run();
  }
}

File: src/main/kotlin/ch/d/peters/zork/Zork.kt
package ch.d.peters.zork

import ch.d.peters.zork.ui.ConsoleUI
import ch.d.peters.zork.application.GameState
import ch.d.peters.zork.application.ZorkActions
import ch.d.peters.zork.application.ZorkLoop
import ch.d.peters.zork.utils.Parser
import ch.d.peters.zork.generators.MapGenerator
import ch.d.peters.zork.generators.PlayerGenerator
import java.util.*


class Zork {
  private val scanner = Scanner(System.`in`)
  private val ui = ConsoleUI()
  private val parser = Parser(scanner)
  private val pGen = PlayerGenerator(scanner, ui)
  private val player = pGen.createPlayer()
  private val mGen = MapGenerator(ui)
  private val rooms = mGen.createRooms()
  private val game = GameState(player, rooms, rooms[3])
  private val actions = ZorkActions(parser, ui, game)
  private val loop = ZorkLoop(actions)

  fun start() {
    loop.run()
  }

  companion object {
    val scanner = Scanner(System.`in`)
    @JvmStatic
    fun main(args: Array<String>) {
      val zork = Zork()
      zork.start()
    }
  }
}
